hr_employee_firstname/models/hr_employee.py

```diff
@@ -90,12 +89,11 @@ def create(self, vals):
     @api.multi
     def write(self, vals):
         if vals.get('firstname') or vals.get('lastname'):
-            vals['name'] = ' '.join([
-                vals.get('firstname') or self.firstname or ' ',
-                vals.get('lastname') or self.lastname or ' ',
-            ])
+            lastname = vals.get('lastname') or self.lastname or ' '
+            firstname = vals.get('firstname') or self.firstname or ' '
+            vals['name'] = self._get_name(lastname, firstname)
         elif vals.get('name'):
-            vals['firstname'], vals['lastname'] = self.split_name(vals['name'])
+            vals['lastname'], vals['firstname'] = self.split_name(vals['name'])
         res = super(HrEmployee, self).write(vals)
         self._update_partner_firstname(self)
```

Comment:
You call _update_partner_firstname in every case ? This makes useless writes if the name of the employee is not changed.